homeassistant/config_entries.py

```diff
@@ -2432,6 +2432,26 @@ def _async_abort_entries_match(
             self._async_current_entries(include_ignore=False), match_dict
         )
 
+    @callback
+    def _abort_if_unique_id_mismatch(
+        self,
+        *,
+        reason: str = "unique_id_mismatch",
+    ) -> None:
+        """Abort if the unique ID does not match the reauth/reconfigure context.
+
+        Requires strings.json entry corresponding to the `reason` parameter
+        in user visible flows.
+        """
+        if (
+            self.source == SOURCE_REAUTH
+            and self._get_reauth_entry().unique_id != self.unique_id
+        ) or (
+            self.source == SOURCE_RECONFIGURE
+            and self._get_reconfigure_entry().unique_id != self.unique_id
```

Comment:
I'm not sure if we should assume the unique ID is being used for detecting these cases.